components/common/TextWithLinks.tsx

```diff
@@ -0,0 +1,31 @@
+// Utility function to detect and replace links with <a> tags
+const wrapLinksInText = (text: string): JSX.Element[] => {
+  // Regular expression to match URLs
+  const urlRegex = /(https?:\/\/[^\s]+)/g
+
+  // Split text into parts, with URLs as matches
+  const parts = text.split(urlRegex)
+
+  return parts.map((part, index) => {
+    if (urlRegex.test(part)) {
```

Comment:
Gotta be a little careful with this because `regexp.test()` is stateful when `regexp` has the global `/g` flag.  Good example here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#using_test_on_a_regex_with_the_global_flag

Though I don't think the global flag is necessary in here, so may be able to just get rid of it.
```suggestion
  const urlRegex = /(https?:\/\/[^\s]+)/

  // Split text into parts, with URLs as matches
  const parts = text.split(urlRegex)

  return parts.map((part, index) => {
    if (urlRegex.test(part)) {
```